internal/expr/expr_test.go

```diff
@@ -307,6 +311,24 @@ AND z = @sqlair_0 -- The line with $Person.id on it
 	"[Bypass[INSERT INTO arr VALUES (ARRAY[[1,2],[] Input[HardMaths.x] Bypass[,4]], ARRAY[[5,6],[] Input[HardMaths.y] Bypass[,8]]);]]",
 	[]any{HardMaths{}},
 	"INSERT INTO arr VALUES (ARRAY[[1,2],[@sqlair_0,4]], ARRAY[[5,6],[@sqlair_1,8]]);",
+}, {
+	"all primitive types",
+	"SELECT (stringCol AS &string), (boolCol AS &bool), (uintCol AS &uint), (uint8Col AS &uint8), (uint16Col AS &uint16), (uint32Col AS &uint32), (uint64Col AS &uint64), (intCol AS &int), (int8Col AS &int8), (int16Col AS &int16), (int32Col AS &int32), (int64Col AS &int64), (float32Col AS &float32), (float64Col AS &float64) FROM t",
+	"[Bypass[SELECT (] Output[[stringCol] [string]] Bypass[), (] Output[[boolCol] [bool]] Bypass[), (] Output[[uintCol] [uint]] Bypass[), (] Output[[uint8Col] [uint8]] Bypass[), (] Output[[uint16Col] [uint16]] Bypass[), (] Output[[uint32Col] [uint32]] Bypass[), (] Output[[uint64Col] [uint64]] Bypass[), (] Output[[intCol] [int]] Bypass[), (] Output[[int8Col] [int8]] Bypass[), (] Output[[int16Col] [int16]] Bypass[), (] Output[[int32Col] [int32]] Bypass[), (] Output[[int64Col] [int64]] Bypass[), (] Output[[float32Col] [float32]] Bypass[), (] Output[[float64Col] [float64]] Bypass[) FROM t]]",
+	[]any{},
+	"SELECT (stringCol AS _sqlair_0), (boolCol AS _sqlair_1), (uintCol AS _sqlair_2), (uint8Col AS _sqlair_3), (uint16Col AS _sqlair_4), (uint32Col AS _sqlair_5), (uint64Col AS _sqlair_6), (intCol AS _sqlair_7), (int8Col AS _sqlair_8), (int16Col AS _sqlair_9), (int32Col AS _sqlair_10), (int64Col AS _sqlair_11), (float32Col AS _sqlair_12), (float64Col AS _sqlair_13) FROM t",
+}, {
+	"custom primitive types",
+	"SELECT name AS &nameStr, address AS &addressStr, id AS &idInt FROM t WHERE name = $nameStr AND address = $addressStr AND id = $idInt",
+	"[Bypass[SELECT ] Output[[name] [nameStr]] Bypass[, ] Output[[address] [addressStr]] Bypass[, ] Output[[id] [idInt]] Bypass[ FROM t WHERE name = ] Input[nameStr] Bypass[ AND address = ] Input[addressStr] Bypass[ AND id = ] Input[idInt]]",
+	[]any{(idInt)(0), (addressStr)(""), (nameStr)("")},
+	"SELECT name AS _sqlair_0, address AS _sqlair_1, id AS _sqlair_2 FROM t WHERE name = @sqlair_0 AND address = @sqlair_1 AND id = @sqlair_2",
+}, {
+	"primitive types passed to prepare unnecessarily",
+	"SELECT id AS &int64, id AS &uint16 FROM t",
+	"[Bypass[SELECT ] Output[[id] [int64]] Bypass[, ] Output[[id] [uint16]] Bypass[ FROM t]]",
+	[]any{int64(0), uint16(0)},
```

Comment:
This is a good point. I don't think we should required that people pass them because they genuinely are special cases, unlike the normal user defined types, passing them would imply you could do weird things like name your string type `int64` and have it work, which I don't think we should allow.

My thinking for having this error is that it is something a user would only ever hit once, then they would know that they don't need to pass the primitive types.  

There is some inconstancy though, we actually let people pass types to `Prepare` that are not used in the query, this error does not fit well with that.

I think I could also change the wording of this error to make it better, how does 
`primitive types do not need to be passed to prepare` sound?

Reply:
I agree with you that we should NOT allow people to pass primitive types to Prepare. I would try to frame the error message to say that you cannot pass types with the name of a primitive one, not that it is not needed, more like not allowed. I also agree with you that we need to be clear that the error is passing it to .Prepare, and not including it in the query string. What do you think of?
```
cannot prepare statement: type samples cannot contain primitive types
```